src/util/git.js

```diff
@@ -114,6 +115,17 @@ export default class Git implements GitRefResolvingInterface {
       };
     }
 
+    // npm local packages are specified as file:, but url parser interprets them as using the file protocol.
+    // This changes the behavior so that git doesn't see this as a hostname, but as a file path.
+    // See #3670.
+    if (parsed.protocol === FILE_PROTOCOL && !parsed.hostname && parsed.path && parsed.port === null) {
+      return {
+        hostname: parsed.hostname,
+        protocol: parsed.protocol,
+        repository: parsed.path,
+      };
+    }
+
     return {
       hostname: parsed.hostname || null,
       protocol: parsed.protocol || 'file:',
```

Comment:
Do you think there is a way we can write a test for this? It doesn't look like there were any unit tests for local git directories using `file:./somewhere/.git` in fact, I didn't even realize that would work until I looked at this PR. It would be really nice to have some tests around that so we don't accidentally break it in the future.